src/models/forecasts.jl

```diff
@@ -2,13 +2,44 @@ abstract type
     Forecast
 end
 
+"""
+    Deterministic
+        A deterministic forecast for a particular data field in a PowerSystemDevice.
+
+"""
 struct Deterministic <: Forecast
-    device::PowerSystemDevice
-    horizon::Int
-    resolution::Dates.Period
-    interval::Dates.Period
-    initialtime::Dates.DateTime
-    data::Dict{Any,Dict{Int,TimeSeries.TimeArray}}
+    device::PowerSystemDevice       # device
```

Comment:
Would it make more sense to store the instance of Deterministic within the PowerSystemDevice?

Reply:
That was the previous implementation, but it didn't allow to make quick changes to the time series data for the same device

Reply:
Clayton and I discussed this a few minutes ago.  Maybe it should be neither.  The device and the forecast don't need to know about each other.  The operation to be executed needs to know about both, so perhaps that's where they should be tied together.

Reply:
One of the key design features in PowerSystems is precisely to make the time series a defining factor in the device specification. For instance, a particular solar plant is defined in part for the availability of the resource a described by a time series or collection of time series

Reply:
The more that I think about this, the more I return to the implementation here. I tried an implementation like:
```
struct Deterministic <: Forecast
    devicename::String              # device
    fieldpath::Array{Symbol,1}      # field of device
    resolution::Dates.Period        # resolution
    initialtime::Dates.DateTime     # forecast availability time
    data::TimeSeries.TimeArray      # TimeStamp - scalingfactor
end
```
and I don't like having the `device` field as a `string` because there could be multiple devices with the same `name::String` in the System. I've returned to the position, despite the somewhat odd and backward reference of having a the `device::PowerDevice` field.

Reply:
I don't see a problem with having the device reference instead of the string. This will avoid issues of duplicate naming.